components/event-bus/internal/knative/util/kn-eventing.go

```diff
@@ -192,27 +201,30 @@ func (k *KnativeLib) UpdateSubscription(sub *evapisv1alpha1.Subscription) (*evap
 
 // SendMessage sends a message to a channel
 func (k *KnativeLib) SendMessage(channel *evapisv1alpha1.Channel, headers *map[string][]string, payload *string) error {
-	httpClient := &http.Client{
-		Transport: initHTTPTransport(),
-	}
+
 	req, err := makeHTTPRequest(channel, headers, payload)
 	if err != nil {
 		log.Printf("ERROR: SendMessage(): makeHTTPRequest() failed: %v", err)
 		return err
 	}
 
-	res, err := httpClient.Do(req)
+	res, err := k.httpClient.Do(req)
 	if err != nil {
 		log.Printf("ERROR: SendMessage(): could not send HTTP request: %v", err)
 		return err
 	}
 	defer func() {
 		_ = res.Body.Close()
+		tran, ok := k.httpClient.Transport.(*http.Transport)
+		if !ok {
+			return
```

Comment:
I think this is more readable:
```
if tran, ok := k.httpClient.Transport.(*http.Transport); ok {
  tran.CloseIdleConnections()
}
```

Reply:
Honestly, the first one is more readable to me :-)

Reply:
I tried the other way but, I got a compilation error as `tran` would be unused var.

Reply:
fixed as agreed